proxy-robots-module/src/main/java/orwell/proxy/Cli.java

```diff
@@ -0,0 +1,86 @@
+package orwell.proxy;
+
+import org.apache.commons.cli.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import orwell.proxy.config.ConfigCli;
+import orwell.proxy.config.EnumConfigFileType;
+
+import java.io.File;
+
+/**
+ * Created by miludmann on 5/5/15.
+ */
+public class Cli {
+    final static protected String DEFAULT_CONFIG_FILEPATH = "/configuration.xml";
+    final static Logger logback = LoggerFactory.getLogger(Cli.class);
+    private String[] args = null;
+    private final Options options = new Options();
+
+    public Cli(String[] args) {
+        this.args = args;
+
+        options.addOption("h", "help", false, "shows help.");
+        options.addOption("f", "file", true, "optional filepath for external configuration file");
+        options.addOption("u", "url", true, "optional url for external configuration file, NOT HANDLED YET");
+    }
+
+    public ConfigCli parse() {
+        CommandLineParser parser = new BasicParser();
+        CommandLine cmd;
+
+        try {
+            cmd = parser.parse(options, args);
+
+            if (cmd.hasOption("h"))
+                help();
+
+            if (cmd.hasOption("f")) {
```

Comment:
Maybe at some point you will want to warn if u and f are both provided at the same time.